poly1_1 prep

PieterTack opened the pull request:
initial commit: fixed some ToDos, added launch vs get_trans_eff test, added function to calculate external PC intersect coordinate

PieterTack committed:
fixed some ToDos, added launch vs get_trans_eff test, added function to calculate external PC intersect coordinate

tschoonj commented:
Looks like a typo

PieterTack committed:
polycap_photon: fixed polyap type

PieterTack committed:
polycap_capil and _photon: several fixes for syntax errors

PieterTack committed:
polycap_photon: adjusted return type of polycap_photon_pc_intersect

PieterTack committed:
defined z_id

PieterTack commented:
Hey Tom,

Hoe kan ik op een fatsoenlijke manier die functie `polycap_photon_pc_intersect` een polycap_vector3 laten returnen, maar tegelijk ook een soort error value laten weergeven? Zowel -1 als NULL gaan uiteraard niet, aangezien dat geen `polycap_vector3` zijn.

Had nu dus als idee om een pointer te laten returnen, maar dan krijg ik (terecht) een waarschuwing dat ik address van lokale variabele aan het returnen ben... Hoe fix ik dit?
Of stel je voor dat ik toch beter een int ofzo laat returnen, en de `polycap_vector3` als pointer argument in de functie plaats en dan zo de waarde aanpas? Kvroeg me af wat als meest elegante oplossing wordt beschouwd.

tschoonj commented on src/polycap-photon.c:
```diff
@@ -348,7 +348,7 @@ polycap_vector3 polycap_photon_pc_intersect(polycap_vector3 photon_coord, polyca
 	phot_temp.z = photon_coord.z + phot_dir.z * (z_fin-photon_coord.z)/phot_dir.z;
 
-	return phot_temp;
+	return &phot_temp;
```
```suggestion
        polycap_vector3 *rv = malloc(sizeof(polycap_vector3));
        *rv = phot_temp;
	return rv;
```

Misschien zo iets? Je zou natuurlijk ook gewoon `phot_temp` kunnen malloc'en in het begin van de functie en die dan zo gebruiken.

Vergeet niet te free'en later!

PieterTack committed:
tests: capil.c adjusted for new polycap_capil_refl_polar syntax

PieterTack committed:
polycap_capil tests: further implementation of theta removal
Current state: tests are botched up. Will adjust input parameters to more realistic case, and eventually remove alfa dependencies

PieterTack committed:
More fixes for tests + preparation to remove alfa dependency from polycap_capil functions

PieterTack committed:
polycap_source: now uses more appropriate simulated open area. Calculated open area printf got botched however

PieterTack committed:
leqk unit tests: fixed

PieterTack commented:
Er is blijkbaar een meson issue. Appveyor werkt normaal wel (op lvserver compiled het en passen de tests)

Ik probeer ook eens te kijken om die LGTM test ook C te laten runnen. Enkel Python is hier niet echt nuttig ;)

tschoonj commented:
Zal dat meson issue fixen. 

LGTM werkt niet omdat de build dependencies vereist waarvan zij op de hoogte moeten gebracht worden door een config file.

PieterTack committed:
polycap_capil: removed alfa as input argument, now calculated automagically from surface norm and photon direction

PieterTack committed:
Polycap_photon: handling of photon start_coords.z > 0 in case of leak_calc

PieterTack committed:
polycap_description: more reliable open area estimation

PieterTack committed:
polycap_source: printing error fixed

PieterTack committed:
polycap.py: adjusted Photon.launch() return values
if photon enters optic: return weights (even if nearly 0)
if photon hits capillary wall: return NULL
if photon gives error or misses optic: returns error

PieterTack commented:
@tschoonj 
Kijk je even polycap.pyx na mbt return values?
Het belangrijke is dat als `polycap_photon_launch` 0 of 1 returned dan raakten de fotonen in het optic, en wordt een bijhorende transmissieefficiencite meegegeven (bij return value 0 zal dit ongeveer 0 zijn)
Bij `polycap_photon_launch` return 2 raakte het foton de capillaire rand, en wordt in `Photon.launch()` NULL returned
Bij -1 en -2 wordt (respecteivelijk error en foton mist optiek) ook NULL returned, maar wordt ook een error/exception gezet die kan opgevangen worden?
Om dan uit meerdere launches de `Source.get_transmission_efficiencies()` na te bootsen, kan dan dus adhv aantal NULL en 0,1 returns bepaald worden wat de open area is, en zou dus hetzelfde resultaat moeten bekomen worden (dit klopt alvast in C code)

Enige waar ik niet zeker van ben is of je default bij return -1 of -2 direct die error ziet, of dat je soms verkeerdelijk het zou kunnen interpreteren als een 2 return (door NULL return).

PieterTack commented:
op mijn TODO lijst voor de nieuwe versie release staat verder nog:

- [x]  rng verplaatsen van `polycap_photon` structure naar `polycap_source` (indien mogelijk, denk het wel)

- [x]  documentatie nog eens nalezen en aanpassen waar nodig (duidelijker stellen dat optic hexagonaal is, Fresnel vergelijkingen aanpassen aan polarisatie-afhankelijkheid, return values van alle functies controleren, ...)

- [x]  LGTM trachten C code te laten controleren

tschoonj commented:
Zie eens naar:

https://github.com/PieterTack/polycap/blob/d795873577c423339cb45e4b412ef18a2eda6650/python/polycap.pyx#L458-L470

Als de error na de call niet NULL is zal er altijd een exception ge-raised worden.
Zoals ge ziet wordt er indien er geen exception geproduceerd is enkel gekeken naar rv == 0: indien waar wordt er None ge-returned. Je zal die conditie moeten veranderen als er iets speciaal moet gebeuren bij rv == 2.

PieterTack committed:
polycap_photon: removed polycap_rng from struct
incorporated in polycap_source, although not really used there either. polycap_rng is only used to obtain random photon start coords, divergency and electric field vectors

PieterTack committed:
python test fixes

PieterTack committed:
Docs: adjusted docs for recent changes

PieterTack commented:
closes issue #55

tschoonj commented:
LGTM is fixed with #57 . See build at https://lgtm.com/logs/b435fa930a69fcc93d8ffdd1454a694a811b4bb1/lang:cpp

PieterTack committed:
example.py: reduce lgtm alerts

PieterTack committed:
docs minor mistake

tschoonj committed:
Fix libpolycap version

tschoonj reviewed (approved):
Merge when 💚
